python/cudf/cudf/core/reshape.py

```diff
@@ -12,6 +12,30 @@
 _axis_map = {0: 0, 1: 1, "index": 0, "columns": 1}
 
 
+def align_objs(objs, how="outer"):
```

Comment:
Since this is used internally and not tested explicitly, we should make it a non-public function (`_align_objs`).

Reply:
@shwina i vaguely remember you creating a similar function to this for aligning indexes of Series passed via a dictionary to DataFrame construction. Where is that function / could it be used here?

Reply:
https://github.com/rapidsai/cudf/blob/b4d76b0dad3ccddc411673eec53c988c8449010f/python/cudf/cudf/core/series.py#L3962

We should remove one of these -- personally I think `reshape` is a better place for something like this to live

Reply:
To make matters worse we also have https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/core/dataframe.py#L5844 :(

Reply:
I think that the `_align_indices` in `dataframe.py` can be left alone. We should just make everything use `cudf.core.reshape._align_objs`